packages/preferences-api/src/browser/preference-proxy.ts

```diff
@@ -26,48 +26,59 @@ export interface PreferenceEventEmitter<T> {
 }
 
 export type PreferenceProxy<T> = Readonly<T> & Disposable & PreferenceEventEmitter<T>;
-export function createPreferenceProxy<T extends Configuration>(preferences: PreferenceService, configuration: T, schema: PreferenceSchema): PreferenceProxy<T> {
+export function createPreferenceProxy<T extends Configuration>(preferences: PreferenceService, schema: PreferenceSchema): PreferenceProxy<T> {
     const toDispose = new DisposableCollection();
     const onPreferenceChangedEmitter = new Emitter<PreferenceChange>();
     toDispose.push(onPreferenceChangedEmitter);
-    toDispose.push(preferences.onPreferenceChanged(e => {
-        if (e.preferenceName in schema.properties) {
-            if (e.newValue) {
+    toDispose.push(preferences.onPreferenceChanged(change => {
+        if (change.preferenceName in schema.properties) {
+            /* Can't simply use change.newValue as this could be a boolean */
+            if (change.newValue !== undefined) {
                 // Fire the pref if it's valid according to the schema
                 if (validatePreference(schema, {
-                    [e.preferenceName]: e.newValue
+                    [change.preferenceName]: change.newValue
                 })) {
-                    onPreferenceChangedEmitter.fire(e);
+                    onPreferenceChangedEmitter.fire(change);
                 } else {
                     // Fire the default preference
                     onPreferenceChangedEmitter.fire({
-                        preferenceName: e.preferenceName,
-                        newValue: configuration[e.preferenceName]
+                        preferenceName: change.preferenceName,
+                        newValue: schema.properties[change.preferenceName].default || undefined
                     });
                 }
             } else {
-                // TODO If it's deleted, fire the default preference
-                onPreferenceChangedEmitter.fire(e);
+                /* Deleted preference, fire the default preference */
+                onPreferenceChangedEmitter.fire({
+                    preferenceName: change.preferenceName,
+                    newValue: schema.properties[change.preferenceName].default || undefined,
+                    oldValue: change.oldValue
+                });
             }
         }
     }));
+
+    /* Create a targer object with only the property names as properties, removing "type" and other stuff from the schema */
+    const targetConfig: Configuration = {};
```

Comment:
why do you need it? just pass `{}` as a target

Reply:
@epatpol https://github.com/theia-ide/theia/pull/1192/files#r166536544 ignore, we have to create `targetConfiguration` and assign at least `undefined` value for each property, otherwise `Object.keys(preference)` won't return anything!

Reply:
Yes but I was under the impression that you wanted to revert the preference-proxy to using configurations AND schemas from your earlier comment above, so that we could go back to the old proxy. Or did I misunderstand when you said to use the default value from the schema inside the config, kinda like so

```
export const filesystemPreferenceSchema: PreferenceSchema = {
    "type": "object",
    "properties": {
        "files.watcherExclude": {
            "description": "List of paths to exclude from the filesystem watcher",
            "default": {
                "**/.git/objects/**": true,
                "**/.git/subtree-cache/**": true,
                "**/node_modules/**": true
            },
            "additionalProperties": {
                "type": "boolean"
            }
        }
    }
};

export interface FileSystemConfiguration {
    'files.watcherExclude': { [globPattern: string]: boolean }
}
export const defaultFileSystemConfiguration: FileSystemConfiguration = {
    'files.watcherExclude': filesystemPreferenceSchema.properties['filesystemPreferenceSchema'].default
};
```

Reply:
I meant that `FileSystemConfiguration` interface should be kept but `defaultFileSystemConfiguration` should be redundant

Reply:
Like the last commit I added? This keeps the preference-proxy mostly the same and just reuse the defaults from the schema.